tests/test_acasclient.py

```diff
@@ -4600,4 +4600,51 @@ def test_018_image_file(self):
+
+    @requires_basic_cmpd_reg_load
+    def test_19_project_restrictions(self):
+        """Check experiment loader project restrictions are working."""
+
+        # Setup for tests
+        # Create a restricted project
+        restricted_project_1 = self.create_basic_project_with_roles()
+        # Create a restricted lot in a different restricted project
+        restricted_project_2 = self.create_basic_project_with_roles()
+        restricted_lot_corp_name = self.create_restricted_lot(restricted_project_2.code_name)
+
+        # (Dissallowed) Restricted experiment proj1 with lot restricted proj2
+        # Create an upload file which uploads to restricted_project_1 but has a lot from restricted_project_2
+        file_to_upload = get_basic_experiment_load_file(self.tempdir, restricted_project_1.names[PROJECT_NAME], corp_name=restricted_lot_corp_name, file_name='uniform-commas-with-quoted-text.csv')
+        # Upload the file and expect it to fail
+        response = self.experiment_load_test(file_to_upload, True, expect_failure=True)
+        # Verify that the error message is correct
+        expected_messages = [
+            {
+                "errorLevel": "error",
+                "message": f"Compounds '{restricted_lot_corp_name}' are in a restricted project that does not match the one entered for this Experiment."
+            }
+        ]
+        self.check_expected_messages(expected_messages, response['errorMessages'])
+
+        # (Allowed) Restricted experiment proj1 with lot restricted proj1
+        # Create another restricted lot in restricted_project_1
+        restricted_lot_corp_name = self.create_restricted_lot(restricted_project_1.code_name)
+        # Create an upload file which uploads to restricted_project_1 and has a lot from restricted_project_1
+        file_to_upload = get_basic_experiment_load_file(self.tempdir, restricted_project_1.names[PROJECT_NAME], corp_name=restricted_lot_corp_name, file_name='uniform-commas-with-quoted-text.csv')
+        # Upload the file and expect it to succeed
+        response = self.experiment_load_test(file_to_upload, True, expect_failure=False)
+
+        # (Allowed) Restricted experiment proj1 with lot unrestricted
+        # Basic cmpdreg load creates a lot in the Global unrestricted project
+        # So we expect that it can be uploaded to a restricted project
+        # Get a file to upload to restricted_project_1 but has a lot from the Global unrestricted project
+        file_to_upload = get_basic_experiment_load_file(self.tempdir, restricted_project_1.names[PROJECT_NAME], corp_name=None, file_name='uniform-commas-with-quoted-text.csv')
+        # Upload the file and expect it to succeed
+        response = self.experiment_load_test(file_to_upload, True, expect_failure=False)
+
+        # (Allowed) Unrestricted experiment proj1 with lot restricted proj2
+        # Create an upload file which uploads to restricted_project_1 but has a lot from restricted_project_2
```

Comment:
From the comment this looks like a good test to run. However I'm confused about your implementation. On the line below this, you seem to be loading a lot in restricted project 1 into experiment restricted project 1.
I don't see you create an unrestricted experiment project or using Global as the experiment project.
Also the lot seems to be from project 1 but your comment here says it's from project 2.

Reply:
I have addressed the comment. Let me know if now the test seems correct to you!

Reply:
Thanks. LGTM.  I changed the test and forgot to fix the comment.